src/hooks/useRescanStatus.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react'
+import { useQuery } from '@tanstack/react-query'
+import { getrescaninfoOptions } from '@/lib/jm-api/generated/client/@tanstack/react-query.gen'
+import { setSession } from '@/lib/session'
+import { useApiClient } from './useApiClient'
+import { useSession } from './useSession'
+
+export const useRescanStatus = () => {
+  const session = useSession()
+  const client = useApiClient()
+
+  const walletFileName = session?.walletFileName
+
+  const { data: rescanInfo, isLoading } = useQuery({
+    ...getrescaninfoOptions({
+      client,
+      path: { walletname: walletFileName || '' },
```

Comment:
What happens when you use `useRescanStatus` when `walletFileName` is not present?

If the component needs this value to be present, it should be a required argument.
e.g. useRescanStatus( { walletFileName })

What do you think?